src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/CreateCaseCallbackService.java

```diff
@@ -1,45 +1,67 @@
 package uk.gov.hmcts.reform.bulkscan.orchestrator.services.ccd;
 
 import com.google.common.base.Strings;
+import com.google.common.collect.ImmutableMap;
+import io.vavr.collection.Array;
 import io.vavr.collection.Seq;
 import io.vavr.control.Either;
 import io.vavr.control.Try;
 import io.vavr.control.Validation;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.bulkscan.orchestrator.client.transformation.InvalidCaseDataException;
+import uk.gov.hmcts.reform.bulkscan.orchestrator.client.transformation.TransformationClient;
 import uk.gov.hmcts.reform.bulkscan.orchestrator.client.transformation.model.request.ExceptionRecord;
 import uk.gov.hmcts.reform.bulkscan.orchestrator.config.ServiceConfigItem;
 import uk.gov.hmcts.reform.bulkscan.orchestrator.services.ccd.callback.CreateCaseValidator;
 import uk.gov.hmcts.reform.bulkscan.orchestrator.services.config.ServiceConfigProvider;
 import uk.gov.hmcts.reform.ccd.client.model.CaseDetails;
 
 import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+import java.util.function.Function;
 
 import static uk.gov.hmcts.reform.bulkscan.orchestrator.services.ccd.CallbackValidations.hasServiceNameInCaseTypeId;
 import static uk.gov.hmcts.reform.bulkscan.orchestrator.services.ccd.EventIdValidator.isCreateCaseEvent;
 
 @Service
 public class CreateCaseCallbackService {
 
+    private static final Logger log = LoggerFactory.getLogger(CreateCaseCallbackService.class);
+
     private final CreateCaseValidator validator;
     private final ServiceConfigProvider serviceConfigProvider;
+    private final TransformationClient transformationClient;
 
     public CreateCaseCallbackService(
         CreateCaseValidator validator,
-        ServiceConfigProvider serviceConfigProvider
+        ServiceConfigProvider serviceConfigProvider,
+        TransformationClient transformationClient
     ) {
         this.validator = validator;
         this.serviceConfigProvider = serviceConfigProvider;
+        this.transformationClient = transformationClient;
     }
 
     /**
      * Create case record from exception case record.
      *
-     * @return Either TBD - not yet implemented
+     * @return Either list of errors or map of changes - new case reference
      */
-    public Either<List<String>, ExceptionRecord> process(CaseDetails caseDetails, String eventId) {
+    public Either<List<String>, Map<String, Object>> process(CaseDetails caseDetails, String eventId) {
```

Comment:
Why is it not going to be an `ExceptionRecord` anymore? Something will still have to prepare it, based on the result from transformation API. This class seems to be the best place for it.